include/fmt/chrono.h

```diff
@@ -759,18 +753,30 @@ inline std::chrono::duration<Rep, std::milli> get_milliseconds(
   return std::chrono::duration<Rep, std::milli>(static_cast<Rep>(ms));
 }
 
-template <typename Rep, typename OutputIt>
+template <typename Char, typename Rep, typename OutputIt>
 OutputIt format_duration_value(OutputIt out, Rep val, int precision) {
-  if (precision >= 0) return format_to(out, "{:.{}f}", val, precision);
-  return format_to(out, std::is_floating_point<Rep>::value ? "{:g}" : "{}",
-                   val);
+  const Char pr_f[] { '{', ':', '.', '{', '}', 'f', '}', 0 };
+  if (precision >= 0) return format_to(out, pr_f, val, precision);
+  const Char fp_f[] { '{', ':', 'g', '}', 0 };
+  const Char format[] { '{', '}', 0 };
+  return format_to(out, std::is_floating_point<Rep>::value ? fp_f : format, val);
 }
 
-template <typename Period, typename OutputIt>
+template <typename Char, typename Period, typename OutputIt>
 OutputIt format_duration_unit(OutputIt out) {
-  if (const char* unit = get_units<Period>()) return format_to(out, "{}", unit);
-  if (Period::den == 1) return format_to(out, "[{}]s", Period::num);
-  return format_to(out, "[{}/{}]s", Period::num, Period::den);
+  if (const char* unit = get_units<Period>()) {
+    string_view s(unit);
```

Comment:
I suggest making `unit` a `string_view` and removing `s`.

Reply:
I'll need to rework some things to do this properly since `get_units` can return a `nullptr`, which fmt's `string_view` doesn't handle properly as far as I know. Since it passes the pointer to `std::char_traits<T>::length()` which will dereference it without checking.

So it either needs to stay as-is, or I can return some other magic value from `get_units` default overload, like an empty string.

Reply:
Let's leave as is this then.